### PR TITLE
feat(self-managed): OpenShift ROSA instructions and rework of the reference arch for OpenShift

### PR DESCRIPTION
**Pull request by leiicamundi**
## Description

This PR revises the installation instructions for Camunda 8 on OpenShift, taking into account versions **8.7** and **8.6**.

- For version **8.7**, the instructions are functional, but the installation process is still broken. This will be fixed during the QA process.  
- This update also adds a reference to the installation of a **ROSA cluster**.  
- Additionally, the instructions for installing Camunda 8 on an OpenShift cluster have been reworked to include **routes as the default exposure solution**.
- Similar to https://github.com/camunda/camunda-docs/pull/4714, we add diagrams

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md
Comment (conceptualshark) [inline docs/self-managed/setup/deploy/openshift/redhat-openshift.md:90]: It might be worth removing this "Prerequisites" heading and allowing the subsequent H5s to be H4s. This nested, we lose the ToC that the additional header could add, and it allows the subsequent headers to stand out a little more.

**Review by conceptualshark — DISMISSED** [review]
Made some small copy edits and left a non-blocking suggestion - I love the reusable structure of these guides, makes the content much easier to read and review!

File: docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md:7]: in what sense keep it aligned?
Maybe we could point out what people should keep aligned?
Structure, implementation details, ...

File: docs/self-managed/setup/deploy/amazon/openshift/assets/rosa-single-region.jpg
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/openshift/assets/rosa-single-region.jpg:1]: Are the VPC cidr ranges correct?
We fixed it on EKS but not sure whether we left the weird cidr ranges for rosa.

Comment (Langleu) [reply docs/self-managed/setup/deploy/amazon/openshift/assets/rosa-single-region.jpg:1]: @leiicamundi as not sure this comment is shown on the overview page.

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:7]: similar question to above on what to keep aligned / consistent between the two.

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:16]: dead link

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:39]: dead link

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:41]: `this guide assumes that the database services (PostgreSQL and OpenSearch) integrated into the default chart will be used`

You mean PostgreSQL and `Elasticsearch` or?

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:89]: can we say for development purposes?

It would also align better with the caution admonition as we recommend in there a separate IAM user.  (one could even go a with a just a role).

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:201]: just a sidenote, strangest requirement ever. Even with the linked documentation it's like why? 😅

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:272]: maybe we could add the link for `official documentation of the module` again

**Review by Langleu — CHANGES_REQUESTED** [review]
Some smaller requests, I only reviewed the next part and would assume 8.6 is the same.

Overall looks quite good 🚀.
I did not try to follow it step by step, let me know if I should run through it all.

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [reply docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:201]: didn't have to 😄 , was just surprised that even the upstream docs didn't give insights to this weird dependency.

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [reply docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:90]: I would probably still say that AWS CLI is for development purposes.
Otherwise it's we recommend separate user for production but we recommend AWS CLI (for what?)

File: docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md
Comment (Langleu) [reply docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md]: thank you 🙇, get it better know what the alignment idea is!

**Review by Langleu — CHANGES_REQUESTED** [review]
Thanks for the quick adjustments, the only thing I'd probably change is the AWS CLI recommendation still

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (Langleu) [reply docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:90]: Or in this example to showcase we're defaulting on AWS CLI as easy usage.

File: docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md
Comment (conceptualshark) [inline docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md:308]:
```suggestion
We strongly recommend managing sensitive information (for example, the OpenSearch or Aurora username and password) using a secure secrets management solution like HashiCorp Vault. For details on how to inject secrets directly into Terraform via Vault, see the [Terraform Vault Secrets Injection Guide](https://developer.hashicorp.com/terraform/tutorials/secrets/secrets-vault).
```

^ This statement was a little unclear to me - I believe the above is what's intended?

**Review by conceptualshark — APPROVED** [review]
Made a few grammar edits and left a small comment regarding wording but otherwise this looks to be in a good place to me. Approving regardless, just let me know if anything else gets updated and this needs a final thumbs up. 👍

**Review by Langleu — APPROVED** [review]

**Review by conceptualshark — DISMISSED** [review]

File: docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md
Comment (akeller) [inline docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md:7]: I wanted to know if we could merge this PR or if we were waiting for an alpha or minor. I didn't see any labels, but I did see this comment was still here.  I'll let the assignee or reviewers decide when to merge.

Comment (Langleu) [reply docs/self-managed/setup/deploy/amazon/amazon-eks/eks-helm.md:7]: My bad, didn't have enough time to review some related changes in other repos.
Running through those as we speak and Léo is back on Friday and would assume him to do the last adjustments then + merge.

The changes would just be references to different branches as those are currently on a feature branch.

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/openshift/redhat-openshift.md:148]: For this one I didn't have enough permissions 🤷 

> Error from server (Forbidden): admission webhook "ingress-config-validation.managed.openshift.io" denied the request: Only privileged service accounts may access

I went for the controller annotation in the end

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/openshift/redhat-openshift.md:65]: would it make sense to call this file `values.yml`?

I called mine `base.yml` as the file was called that way.

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/openshift/redhat-openshift.md:329]: nice, just as a reminder that we have to change it back before the 8.7 release but since it's in the repo we can easily switch it up 🚀

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/openshift/redhat-openshift.md:271]: learned the hard way about the difference ^^, wish there was more Helm chart feedback as it initially looks successful but then nothing is deployed.

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/openshift/redhat-openshift.md:308]: in regards to postgres, I wonder whether the helm chart has a default if so whether we should use it or is empty a totally acceptable value?

File: docs/self-managed/setup/deploy/openshift/redhat-openshift.md
Comment (Langleu) [inline docs/self-managed/setup/deploy/openshift/redhat-openshift.md:353]: sadly caught a broken snapshot release and Zeebe wasn't coming up 😓

**Review by conceptualshark — APPROVED** [review]

**Pull request merged**
